### PR TITLE
Add README to propagators/aws/xray

### PR DESCRIPTION
**Pull request by bryan-aguilar**
This PR adds a basic README to the x-ray package which highlights possible incompatibilities between the `x-ray IDGenerator` and the `traceIDRatioSampler`.

### CODE REVIEW

File: propagators/aws/xray/README.MD [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# AWS X-Ray Propagator/IDGenerator
+
+This package contains an AWS x-ray compatible propagator along
```
Comment (Aneurysm9) [inline propagators/aws/xray/README.MD:3]:
```suggestion
This package contains an AWS X-Ray compatible propagator along
```

Carry throughout.

File: propagators/aws/xray/README.MD [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+with an x-ray compatible IDGenerator.
+
+## `traceIdRatioSampler` and `x-ray IDGenerator` compatability
+
+It is a general suggestion to `NOT` use the `traceIDRatioSampler` while also
+using the `x-ray IDGenerator`. The non random nature of building an x-ray traceId
```
Comment (Aneurysm9) [inline propagators/aws/xray/README.MD:9]:
```suggestion
using the `x-ray IDGenerator`. The non-random nature of building an x-ray traceId
```

File: propagators/aws/xray/README.MD [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+may lead to inconsistent sampling results.
```
Comment (Aneurysm9) [inline propagators/aws/xray/README.MD:10]:
```suggestion
may lead to unexpected sampling results.
```

```suggestion
may lead to overly consistent sampling results.
```

???

File: CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Added
 
 - Consistent probability sampler implementation. (#1379)
+- Add README to `go.opentelemetry.io/contrib/propagators/aws/xray`. (#2321)
```
Comment (dmathieu) [inline CHANGELOG.md:16]: I wouldn't have added a changelog entry for a readme addition/change. It's not technically something meaningful for folks who upgrade from a previous version.

File: propagators/aws/xray/README.MD [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# AWS X-Ray Propagator/IDGenerator
+
+This package contains an AWS X-Ray compatible propagator along
+with an x-ray compatible IDGenerator.
+
+## `traceIdRatioSampler` and `x-ray IDGenerator` compatability
+
+It is a general suggestion to `NOT` use the `traceIDRatioSampler` while also
+using the `x-ray IDGenerator`. The non-random nature of building an x-ray traceId
+may lead to unexpected sampling results.
```
Comment (dmathieu) [inline propagators/aws/xray/README.MD:10]: How about adding a `Usage` section?

File: propagators/aws/xray/README.MD [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# AWS X-Ray Propagator/IDGenerator
+
+This package contains an AWS X-Ray compatible propagator along
+with an x-ray compatible IDGenerator.
+
+## `traceIdRatioSampler` and `x-ray IDGenerator` compatability
```
Comment (dmathieu) [inline propagators/aws/xray/README.MD:6]:
```suggestion
## `traceIdRatioSampler` and `x-ray IDGenerator` compatibility
```

File: CHANGELOG.md (hunk 4 above)
Comment (bryan-aguilar) [reply CHANGELOG.md]: Noted, that's a good point. Removed it.

File: propagators/aws/xray/README.MD (hunk 5 above)
Comment (bryan-aguilar) [reply propagators/aws/xray/README.MD:10]: I think a `usage` section would be useful but also don't want to hold back this PR as I may not have time to get to it immediately. I believe adding the warning for the incompatibilities between sampler/generator warrant this going through. If desired, I can open an issue to expand the README with a usage section.

File: propagators/aws/xray/README.MD [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# AWS X-Ray Propagator/IDGenerator
+
+This package contains an AWS X-Ray compatible propagator along
+with an x-ray compatible IDGenerator.
```
Comment (MrAlias) [inline propagators/aws/xray/README.MD:4]:
```suggestion
This package contains an AWS X-Ray compatible `TextMapPropagator` and `IDGenerator`.
```

**Review by hanyuancheung — APPROVED** [review]

**Pull request merged**